data-prepper-plugins/s3-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorkerIT.java

```diff
@@ -0,0 +1,219 @@
+                additionalEventAssertions.accept(eventRecord.getData());
+            }
+            recordsReceived += recordsCollection.size();
+            return null;
+        })
+                .when(buffer).writeAll(anyCollection(), anyInt());
+    }
+
+    private S3SelectObjectWorker createObjectUnderTest(final int numberOfRecordsToAccumulate,
+                                                       final CompressionType compressionType,
+                                                       final RecordsGenerator recordsGenerator,
+                                                       final String expression) {
+        S3SelectCSVOption s3SelectCSVOption = mock(S3SelectCSVOption.class);
+        when(s3SelectCSVOption.getFileHeaderInfo()).thenReturn("none");
+        final S3ObjectRequest request = new S3ObjectRequest.Builder(buffer, numberOfRecordsToAccumulate,
+                Duration.ofMillis(TIMEOUT_IN_MILLIS), s3ObjectPluginMetrics)
+                .bucketOwnerProvider(bucketOwnerProvider)
+                .s3SelectCSVOption(s3SelectCSVOption)
+                .s3SelectJsonOption(new S3SelectJsonOption())
+                .s3AsyncClient(s3AsyncClient)
+                .eventConsumer(eventMetadataModifier).expressionType("SQL")
+                .serializationFormatOption(S3SelectSerializationFormatOption.valueOf(
+                        recordsGenerator.getFileExtension().toUpperCase()))
+                .expression(expression)
+                .compressionType(compressionType).
+                s3SelectResponseHandlerFactory(new S3SelectResponseHandlerFactory()).build();
+        return new S3SelectObjectWorker(request);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(IntegrationTestArguments.class)
+    void parseS3Object_correctly_loads_data_into_Buffer(
+            final RecordsGenerator recordsGenerator,
+            final int numberOfRecords,
+            final int numberOfRecordsToAccumulate,
+            final boolean shouldCompress) throws Exception {
+
+        final String key = getKeyString(recordsGenerator, numberOfRecords, shouldCompress);
+        final CompressionType compressionType = shouldCompress ? CompressionType.GZIP : CompressionType.NONE;
+        final S3SelectObjectWorker objectUnderTest = createObjectUnderTest(numberOfRecordsToAccumulate,
+                compressionType,
+                recordsGenerator,
+                recordsGenerator.getS3SelectExpression());
+
+        s3ObjectGenerator.write(numberOfRecords, key, recordsGenerator, shouldCompress);
+        stubBufferWriter(recordsGenerator::assertEventIsCorrect, key);
+
+        parseObject(key, objectUnderTest);
+
+        final int expectedWrites = numberOfRecords / numberOfRecordsToAccumulate + (numberOfRecords % numberOfRecordsToAccumulate != 0 ? 1 : 0);
+
+        verify(buffer, times(expectedWrites)).writeAll(anyCollection(), eq(TIMEOUT_IN_MILLIS));
+
+        assertThat(recordsReceived, equalTo(numberOfRecords));
```

Comment:
Is it possible to assert on the contents of the records too?

Reply:
Thanks for your review Krishna, assert event data already validating in the RecordsGenerator implementations. so i think we no need to verify explicitly here. Please suggest.

data-prepper-plugins/s3-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/source/ParquetRecordsGenerator.java
Line No: 45
Method Name: assertEventIsCorrect(final Event event)